File: lecture_006(2DArray)/l_010_searchinsorted2darray.java
import java.util.Scanner;
public class l_010_searchinsorted2darray {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        int n = scn.nextInt();
        int[][] arr = new int[n][n];
        input(arr);
        int data = scn.nextInt();
        searchInSorted2dArray(arr,data);
    }
    public static void display(int[][] arr) {
        for(int i=0; i<arr.length; i++) {
            for(int j=0; j<arr[i].length; j++) {
                System.out.print(arr[i][j]+" ");
            }
            System.out.println();
        }
    }
    public static void input(int[][] arr) {
        for(int i=0; i<arr.length; i++) {
            for(int j=0; j<arr[i].length; j++) {
                arr[i][j] = scn.nextInt();
            }
        }
        display(arr);
    }
    public static void searchInSorted2dArray(int[][] arr, int data) {
        int i = 0;
        int j = arr[0].length-1;

        while(i<arr.length && j>=0) {
            if(data<arr[i][j]) {
                j--;
            }
            else if(data>arr[i][j]) {
                i++;
            }
            else {
                System.out.println(i+","+j);
                return;
            }
        }
        System.out.println("Not Found");
    }
}

File: lecture_006(2DArray)/l_010_searchinsorted2darray.cpp
#include<iostream>
#include<vector>
using namespace std;

void display(vector<vector<int>> &arr) {
    for(int i=0; i<arr.size(); i++) {
        for(int j=0; j<arr[0].size(); j++) {
            cout<<arr[i][j]<<" ";
        }
        cout<<endl;
    }
}
void input(vector<vector<int>> &arr) {
    for(int i=0; i<arr.size(); i++) {
        for(int j=0; j<arr[0].size(); j++) {
            cin>>arr[i][j];
        }
    }
    display(arr);
}
void searchinsorted2darray(vector<vector<int>> &arr, int data) {
    int i = 0;
    int j = arr[0].size()-1;

    while(i<arr.size() && j>=0) {
        if(data<arr[i][j]) {
            j--;
        }
        else if(data>arr[i][j]) {
            i++;
        }
        else {
            cout<<i<<","<<j<<endl;
            return;
        }
    }
    cout<<"Not Found"<<endl;
}
int main() {
    int n;
    cin>>n;
    vector<vector<int>>arr(n,vector<int>(n,0));       //vector<int>arr(n,0);
    input(arr);
    int data;
    cin>>data;
    searchinsorted2darray(arr,data);
}